apps/api/src/services/sendgrid_handler.py

```diff
@@ -0,0 +1,72 @@
+# using SendGrid's Python Library
+# https://github.com/sendgrid/sendgrid-python
+import os
+from logging import getLogger
+from typing import Iterable, Optional, Tuple, TypedDict, Union
+
+import aiosendgrid
+from httpx import HTTPStatusError
+from sendgrid.helpers.mail import Email, Mail, Personalization
+
+log = getLogger(__name__)
+
+SENDGRID_API_KEY = os.getenv("SENDGRID_API_KEY")
+
+PersonalizationData = TypedDict(
+    "PersonalizationData",
+    {
+        "email": str,
+        "first_name": Optional[str],
+        "last_name": Optional[str],
+        "passphrase": Optional[str],
+    },
+)
```

Comment:
Chore: This is using an older syntax which has limited support. We can use the class-based syntax
We can also use the `NotRequired` qualifier instead of optional to indicate it does not need to be provided.
```python
class PersonalizationData(TypedDict):
    email: str
    first_name: NotRequired[str]
    ...
```

Reply:
Shoot, I was looking for something like this. I'll implement this.